x/incentives/keeper/suite_test.go

```diff
@@ -196,8 +194,8 @@ func (suite *KeeperTestSuite) setupNewGaugeWithDenom(isPerpetual bool, coins sdk
 }
 
 // SetupNewGaugeWithDenom creates a gauge with the specified duration and denom.
-func (suite *KeeperTestSuite) SetupNewGaugeWithDenom(isPerpetual bool, coins sdk.Coins, denom string) (uint64, *types.Gauge, sdk.Coins, time.Time) {
-	return suite.setupNewGaugeWithDenom(isPerpetual, coins, defaultLockDuration, denom)
+func (suite *KeeperTestSuite) SetupNewGaugeWithDenom(isPerpetual bool, coins sdk.Coins, denom string, numEpochsPaidOver uint64) (uint64, *types.Gauge, sdk.Coins, time.Time) {
+	return suite.setupNewGaugeWithDenom(isPerpetual, coins, defaultLockDuration, denom, numEpochsPaidOver)
```

Comment:
This feels odd to me, num epochs paid over only makes sense as an arg if `isPerpetual=false`

Perhaps make a new constructor for `SetupNewNonPerpetualGauge`?

Reply:
yes, I'll do it

Reply:
can you check this [issue](https://github.com/osmosis-labs/osmosis/issues/2327) and tell if this a problem or not